### DISCUSSION TITLE
How to create different applications

### DISCUSSION THREAD

**Discussion by ruben-bc** [InterDigitalInc/AdvantEDGE#346] [Q&A] [answered]
Hello all,

Once I did all the demos (1, 2, 3 and first scenario), I want to make another application that hasn't been done yet, like for example launch web server in the platform or any other similar case that can be possible to do. After read all the documentation, it isn't clear for me how can I do this, because the examples in the guide are with iperf or with the APIs of MEC011 and MEC012. I don't know how to manage the APIs in case these are necessary for the case I mention.

Thank you so much in advance.

**Comment by dilallkx** [comment, marked as answer]
Hi @rubenbc28,

Thanks again for reaching out!

The demo applications were developed to showcase key platform features & edge services. Demo1 & demo2 use iperf container images to showcase network characteristics emulation. They also provide an example service to demonstrate the location & application state transfer services. Demo3 showcases the latest MEC services MEC011 & MEC021.

These demo applications are provided as examples and can be used as a baseline for creating new scenarios & services.

The platform usage workflow can be found here:
https://interdigitalinc.github.io/AdvantEDGE/docs/usage/usage-workflow/

For your example of running a web server you can use the following procedure:
- Either create your own web server container image or simply use one from a public registry (e.g. nginx in dockerhub)
- Create & configure a scenario using the AdvantEDGE frontend
  - You can start from the demo scenarios or create a new one from scratch
  - When you configure your application, set the image name to your webserver container image
  - To expose your service, set the necessary ports in the configuration pane
- Save & deploy your scenario in your sandbox
- Access your service once the pod is up & running
- Trigger any events & run your tests
- Terminate your scenario

Details on how to configure a scenario using the AdvantEDGE frontend can be found here:
https://interdigitalinc.github.io/AdvantEDGE/docs/usage/gui/config-view/

I hope this helps you get started and that the documentation proves useful in explaining key features & controls.
Best regards!

**Reply by ruben-bc** [reply]
Hi Kevin,

All this information will undoubtedly be of great help, especially for what wasn't clear to me. I'll follow the steps you tell me and try to get the desired application. I hope to do well!

Thanks for the details and the quick response.

Regards.
